HPXMLtoOpenStudio/resources/hpxml.rb

```diff
@@ -3525,15 +3530,23 @@ def to_doc(building)
       if (not @exterior_shading_type.nil?) || (not @exterior_shading_factor_summer.nil?) || (not @exterior_shading_factor_winter.nil?)
         exterior_shading = XMLHelper.add_element(window, 'ExteriorShading')
         sys_id = XMLHelper.add_element(exterior_shading, 'SystemIdentifier')
-        XMLHelper.add_attribute(sys_id, 'id', "#{id}ExteriorShading")
+        if @exterior_shading_id.nil?
+          XMLHelper.add_attribute(sys_id, 'id', "#{id}ExteriorShading")
+        else
+          XMLHelper.add_attribute(sys_id, 'id', @exterior_shading_id)
```

Comment:
Same for interior/exterior shading. We need this, otherwise the `id` gets overwritten.